ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/BodyValidator.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright contributors to Hyperledger Besu.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.mainnet;
+
+import org.hyperledger.besu.datatypes.Hash;
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+import org.hyperledger.besu.ethereum.core.Request;
+import org.hyperledger.besu.ethereum.core.Transaction;
+import org.hyperledger.besu.ethereum.core.TransactionReceipt;
+import org.hyperledger.besu.ethereum.core.Withdrawal;
+import org.hyperledger.besu.evm.log.LogsBloomFilter;
+
+import java.util.List;
+
+/** A utility class for body validation tasks. Implemented utilising BodyValidation */
+public class BodyValidator {
```

Comment:
Being an instantiated class means it can be easily mocked in tests, so we can test code that utilises it without having to supply valid data and deal with real hashes.

Reply:
Whilst it's not ideal to deal with real hashes it's not difficult to work with in the tests. Similar @siladu comment don't want to have two classes that can be used for calculating these hashes. My preference would be remove this class.

Reply:
Alright, I've updated it to use real hashes instead